### PR TITLE
Added support for Eros by PJB

### PR DESCRIPTION
**Pull request by 602studios**
## Description

Added the base files for the Eros keyboard PCB.

These include:
config.h
eros.c
eros.h
info.json
rules.mk
readme.md
keymaps/default/keymap.c
keymaps/via/config.h
keymaps/via/keymap.c
keymaps/via/rules.mk

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

N/A

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/pjb/eros/rules.mk [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = bootloadHID
```
Comment (fauxpark) [inline keyboards/pjb/eros/rules.mk:5]: Is this correct?

File: keyboards/pjb/eros/rules.mk [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
```
Comment (fauxpark) [inline keyboards/pjb/eros/rules.mk:16]:
```suggestion
```

File: keyboards/pjb/eros/rules.mk [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+BACKLIGHT_ENABLE = no     # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
```
Comment (fauxpark) [inline keyboards/pjb/eros/rules.mk:18]:
```suggestion
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Enable audio output
```

File: keyboards/pjb/eros/rules.mk [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+WS2812_DRIVER = i2c
+VIA_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/pjb/eros/rules.mk:20]:
```suggestion
```
VIA must not be enabled at keyboard level.

File: keyboards/pjb/eros/info.json [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        },
+		
+	  "LAYOUT_tkl_ansi_f13_Split_Shift": {
```
Comment (fauxpark) [inline keyboards/pjb/eros/info.json:16]: Layout macros must be all lowercase after the `LAYOUT_`.

File: keyboards/pjb/eros/rules.mk [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+BACKLIGHT_ENABLE = no     # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Enable audio output
+WS2812_DRIVER = i2c
```
Comment (zvecr) [inline keyboards/pjb/eros/rules.mk:18]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Comment by 602studios** [comment]
Uploads from 4/3/22 did not change any of the files! They should all be the same version as reviewed.

File: keyboards/pjb/eros/readme.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+* Keyboard Maintainer: [Phil](https://github.com/602studios)
+* Hardware Supported: Eros TKL PCB
+* Hardware Availability: [Geekhack Group Buy]( https://geekhack.org/index.php?topic=115569.0)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make pjb/eros/default
+
+Flashing example for this keyboard ([after setting up the bootloadHID flashing environment](https://docs.qmk.fm/#/flashing_bootloadhid))
```
Comment (fauxpark) [inline keyboards/pjb/eros/readme.md:13]:
```suggestion
Flashing example for this keyboard:
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/pjb/eros/keymaps/via/config.h [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define DYNAMIC_KEYMAP_LAYER_COUNT 4
```
Comment (fauxpark) [inline keyboards/pjb/eros/keymaps/via/config.h:19]: This is the default, this file can be removed.

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**